components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/InMemoryAPIDeployer.java

```diff
@@ -200,16 +236,22 @@ public GatewayAPIDTO getAPIArtifact(String apiId, String gatewayLabel) {
      *@param tenantDomain - Tenant Domain of the API
      * @return Map that contains the UUID and label of the API
      */
-    public Map <String, String> getGatewayAPIAttributes(String apiName, String version, String tenantDomain) {
+    public Map <String, String> getGatewayAPIAttributes(String apiName, String version, String tenantDomain)
+            throws  ArtifactSynchronizerException {
         Map<String, String> apiAttributes = null;
         if (artifactRetriever != null) {
             try {
                 apiAttributes = artifactRetriever.retrieveAttributes(apiName, version, tenantDomain);
+                log.debug("API Attributes retrieved for " + apiName + "  from storage");
```

Comment:
Better to check if debug is enabled.

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/9546